### PR TITLE
add changes for flat encoding

### LINKED ISSUE #7522: Implement flat-encoding (pectra-devnet-4)
EIP-7251: max eb, change request to flat encoding https://github.com/ethereum/EIPs/pull/8857
EIP-6110: deposits, change request to flat encoding https://github.com/ethereum/EIPs/pull/8856
EIP-7002: withdrawals, change request to flat encoding https://github.com/ethereum/EIPs/pull/8855
EIP-7685: change requests hash to flat hash
https://github.com/ethereum/EIPs/pull/8854

### PR DESCRIPTION
**Pull request by rjnrohit**
Resolves #7522

_Please choose one of the keywords above to refer to the issue this PR solves followed by the issue number (e.g. Fixes #000). If no issue number, remove the line. Also, remove everything marked optional that is not applicable. Remove this note after reading._

## Changes

- _List the changes_

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [ ] No

#### Notes on testing

_Optional. Remove if not applicable._

#### Requires documentation update

- [ ] Yes
- [ ] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [ ] No

_If yes, fill in the details here. Remove if not applicable._

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

**Review by LukaszRozmej — COMMENTED** [review]
So many allocations!

**Comment by rjnrohit** [comment]
> So many allocations!

Requesting for re-review, addressed this issue in the latest commits

**Review by LukaszRozmej — COMMENTED** [review]
Why not just use RlpDecoders?

**Comment by rjnrohit** [comment]
> Why not just use RlpDecoders?

Requests will be  encoded as `request_type` prepended to an opaque byte array `request_data`. So I'm rlp decoding a list of byte arrays then decoding a request from the byte array

**Comment by LukaszRozmej** [comment]
> > Why not just use RlpDecoders?
> 
> https://eips.ethereum.org/EIPS/eip-7685#execution-layer Requests will be encoded as `request_type` prepended to an opaque byte array `request_data`. So I'm rlp decoding a list of byte arrays then decoding a request from the byte array

Sounds like something our RlpDecoder infrastructure could support, add decoder for external list, and then call internal decoders for decoding items dependent on type. No byte array allocation needed, everything done on one pooled buffer through NettyRlpStream

File: src/Nethermind/Nethermind.Serialization.Rlp/BlockDecoder.cs [hunk 1]
```diff
@@ -116,16 +116,14 @@ public class BlockDecoder : IRlpValueDecoder<Block>, IRlpStreamDecoder<Block>
 
                 if (lengthWasRead)
                 {
-                    int requestsLength = rlpStream.ReadSequenceLength();
-                    int requestsCheck = rlpStream.Position + requestsLength;
+                    // read an array of byte arrays
+                    byte[][] requestsEncoded = rlpStream.DecodeByteArrays();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Serialization.Rlp/BlockDecoder.cs:120]: unneeded, make it lazy, call internal decoders based on type and progressing the stream

**Comment by MarekM25** [comment]
PR description!

**Pull request closed without merging**